pkg/proxy/proxy.go

```diff
@@ -143,6 +159,13 @@ func (p *proxy) invokeGrpc(writer http.ResponseWriter, request *http.Request) {
 		writeError(writer, errors.New("expected EOF"))
 		return
 	}
+	if status, ok := outputSignal.GetData().Headers[XHttpStatusHeader]; ok {
+		code, err := strconv.Atoi(status)
+		if err != nil {
+			writeError(writer, fmt.Errorf("Invalid status code %q", status))
+		}
+		writer.WriteHeader(code)
```

Comment:
Should move inside `writeError()` as an additional parameter IMO

Reply:
why? it's not an error

Reply:
So that writeHeader is used in a single place. Has this been tested out of the tests using mocks? I remember seeing with Florent that calling writeHeader too late silently fails (which the mocks used in the tests don't showcase)

Reply:
I removed an ineffective call to writeError below. With this flow WriteHeader will only be called once per response. I don't think it should be contained within the writeError function since it's not and error, and unlike the calls to writeError, there is expected to be further work and not immediately returning.